scanpy/tests/test_plotting.py

```diff
@@ -90,6 +90,8 @@ def test_heatmap(image_comparer):
     )
     save_and_compare_images('master_heatmap_std_scale_var')
 
+    # test standard_scale_obs and groupby = index
+    adata.obs.set_index('cell_type', inplace=True)
```

Comment:
This case looks really weird to me, making the index very non-unique is a bit of an anti-pattern to me.

Could this be tested more directly with something like:

```
subset = adata[:10].copy()
subset.obs_names.name = "index"
sc.pl.heatmap(
    subset
    ...
    groupby="index",
    ....,
)
```

I also don't love that if a case gets added after this, the `adata` would be in a weird state. Maybe this test should get divided up into separate cases?

If you want to check that non-unique index values get grouped together, could that test look more like:

```python
adata = AnnData(
    np.arange(16).reshape(4, 4),
    obs=pd.DataFrame(
        {"cat": list("aabb")},
         index=pd.Index([f"cell-{i}" for i in range(4)], name="cell_id")
    )
)
adata_nonuniq_index = adata.copy()
adata_nonuniq_index.obs.set_index("cat", inplace=True)
```

And then check that plots using either are equivalent? This wouldn't need to add any more reference images. If you like #1587, that can get merged pretty quick and you can use the `check_same_image` fixture for this.

Reply:
I added a new test but I didn't use yet the new `check_same_image` as I noticed after the fact that this was already merged into master.